### DISCUSSION TITLE
Is there a way to to skip the build cache in Vercel?

### DISCUSSION THREAD

**Discussion by Blatty** [vercel/vercel#8057] [Help] [answered]
Hey there,

I'm trying to find out if there's a flag or environment variable to skip the build cache in Vercel? Similar to how it's used in the CLI: https://vercel.com/docs/cli#introduction/global-options/force

**Comment by amyegan** [comment, marked as answer]
Hi @Blatty. We don't currently have a way to disable the build cache for automated deployments using Vercel for Git integrations, but we do have a few other ways to deploy without build cache:
- [Redeploying](https://vercel.com/docs/concepts/deployments/redeploying) from the Dashboard will bypass the Build Cache, unless you check the box to include it
- Using the [API](https://vercel.com/docs/rest-api#introduction/api-basics) using the `forceNew` query parameter to [create a new deployment](https://vercel.com/docs/rest-api#endpoints/deployments/create-a-new-deployment)
- Using the [CLI](https://vercel.com/docs/cli#introduction/global-options/force) with `--force` option, which you already mentioned

**EDIT (14th Oct 2022):** The `VERCEL_FORCE_NO_BUILD_CACHE` env var now exists! Set this to `1` and the build cache will be skipped.

**Reply by Blatty** [reply]
Thank you for the reply @amyegan . We're only using the Git <> Vercel integration though, there's no way to skip the build cache from there? Or is it planned in any near future?

e.g: `VERCEL_SKIP_BUILD_CACHE=1` ?

We have quite a big monorepo and they all use the Git integration, one project is using symlinks and it doesn't seem to work when the build cache hits:

```node_modules/@acme/utils is a symlink and it's children will be excluded from cache```

related issue: #8063

**Reply by amyegan** [reply]
Sorry @Blatty 😞 There is currently no way to skip the build cache from the Vercel Git integration. Some people disable automatic deployment in `vercel.json` and use Vercel CLI with GitHub Actions to set up automatic deployments instead. That would give you more control to define how and when your project is deployed.

**Reply by Pieparker** [reply]
Looping back here to share that the `VERCEL_FORCE_NO_BUILD_CACHE` env var now exists! Set this to `1` and the build cache will be skipped.

**Reply by NavidK0** [reply]
Excellent addition guys! It seems however that with VERCEL_FORCE_NO_BUILD_CACHE enabled, the build cache is still populated at the end of the build process. Is that a bug or intentional?

Beginning the deployment
```
Retrieving list of deployment files...
VERCEL_FORCE_NO_BUILD_CACHE is set so skipping build cache step
Downloading 6030 deployment files...
```

End of the deployment
```
Build completed. Populating build cache...
Uploading build cache [697.59 MB]...
Build cache uploaded: 13.731s
```

Thanks for all your hard work!

**Reply by amyegan** [reply]
I think populating the cache is just part of the standard build process. So the cache is not used for the build, as you saw in the beginning log snippet, but the build cache is populated at the end. This would allow you to use it for a future build if you chose, and it might possibly be useful for other things I'm not thinking of.

**Reply by NavidK0** [reply]
Okay, cool. I was just wondering. Thanks for the quick response!

**Reply by gf5901** [reply]
I have `VERCEL_FORCE_NO_BUILD_CACHE` set to 1, but it is still caching the builds. Any ideas?

**Reply by amyegan** [reply]
Hi @gf5901! With `VERCEL_FORCE_NO_BUILD_CACHE` enabled, you should see `VERCEL_FORCE_NO_BUILD_CACHE is set so skipping build cache step` near the top of the build log to indicate that the cache is not being used. Otherwise, you should see `Restored build cache` near the top of the build log to indicate that the cache is being used. Can you share more about what you're seeing in the build log for your deployment?

**Reply by gf5901** [reply]
So I see `VERCEL_FORCE_NO_BUILD_CACHE is set so skipping build cache step` at the top of the log, but then later on in the log I see: 
```
• Remote computation caching enabled
@unirepo/web:build: cache hit, replaying output 2e0f3c1bfab1a4b7
```
And it uses the cached build. Then I have to manually redeploy it for it to not use the cached build. Not sure if it's something I configured incorrectly on my side or not, but I'd like it to not cache anything when it deploys to production since production uses different environment variables from my staging environment.

**Reply by amyegan** [reply]
Do you have a deployment link, public repo, or [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share?

**Reply by gf5901** [reply]
I do not, should I add you to the project so you can view the deployment link? It's been quite troublesome for me. If I forget to manually redeploy with the build cache turn off, all our users start hitting our QA environment. Is there anyway I can set it to manually kick off builds to production instead of on changes to the main branch?

**Reply by Pieparker** [reply]
Hey gf5901! This sounds like Remote Caching is enabled. This is [expected by default](https://vercel.com/docs/concepts/monorepos/remote-caching#use-remote-caching-during-vercel-build) when Turborepo is enabled and isn't controlled by the build cache; they're autonomous.

The above link shows how and where Remote Caching can be toggled within the UI. I do hope that the helps!

**Reply by gf5901** [reply]
Thanks a ton, @Pieparker! Looks like turning off Remote Caching fixed the issue.

**Comment by johnlindquist** [comment]
@amyegan Is the `forceNew` query parameter supported when using deploy hooks?

**Reply by deve-sh** [reply]
You can use the `buildCache` query param to do so: https://vercel.com/docs/deployments/deploy-hooks#build-cache
